app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -84,4 +110,18 @@ def clear_session_data
     user_session.delete(unconfirmed_phone_session_key)
     user_session.delete(confirmation_code_session_key)
   end
+
+  def sms_enabled?
```

Comment:
I haven't tested this, so I could be wrong, but I think using classes to handle this would also allow us to remove the dependency on the `send_two_factor_authentication_code`, and therefore would not require modifying the upstream gem to support options since the logic would be contained in the classes, and wouldn't have to go through the gem.